playbook.yml

```diff
@@ -20,70 +20,46 @@
         - python-pip
         - unattended-upgrades
 
+  - name:    Add Docker GPG key
+    apt_key: url=https://download.docker.com/linux/ubuntu/gpg
+
+  - name:    Add Docker APT repository
+    apt_repository:
+      repo: deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable
+
+  - name:    Install Docker
+    apt:
+      name: docker-ce
+      state: present
+      update_cache: yes
+
   - name: Disallow password authentication
     lineinfile: dest=/etc/ssh/sshd_config
                 regexp="^PasswordAuthentication"
                 line="PasswordAuthentication no"
                 state=present
     notify: Restart ssh
 
-  - name: checkout the web-platform-tests
-    git:
-      repo: 'https://github.com/web-platform-tests/wpt.git'
-      dest: ~/web-platform-tests
-      force: yes
-
-  - name: make python virtualenv
-    pip:
-      name: virtualenv
-      virtualenv: ~/web-platform-tests/_venv
-
-  - name: ensure wpt service is configured
+  - name: ensure wpt docker service is configured
     template:
-      src: files/wpt.service
+      src: files/docker.wpt.service
       dest: /etc/systemd/system/
 
-  - name: ensure config.json is in the correct location
-    template:
-      src: files/config.json
-      dest: /root/web-platform-tests/
-
-  - name: Collect list of host aliases
-    shell: /root/web-platform-tests/wpt make-hosts-file
-    register: hosts_file_contents
-
-  - name: Update the hosts file
-    lineinfile:
-      line: '{{item}}'
-      dest: /etc/hosts
-    with_items: '{{hosts_file_contents.stdout_lines}}'
-
   - name: Enable the wpt service in systemd
     systemd:
       enabled: True
-      name: wpt
+      name: docker.wpt
       state: started
 
   - include: certbot.yml
     when: inventory_hostname != 'vagrant'
 
   - name: Add cron job to sync wpt
     cron:
-      name: Synchronize WPT
-      job: "cd /root/web-platform-tests/ && git pull origin master && systemctl restart wpt"
-      minute: 0
-      hour: 23
-
+      name: Redeploy a new docker service every minute
+      job: "docker pull bmac/wpt && docker service update --image bmac/wpt:latest wpt-service"
```

Comment:
`docker service update --image bmac/wpt:latest wpt-service` is the line that tells docker service to do a 0 downtime redeploy. We will probably want to only do this if the `docker pull` finds a new version of the wpt docker instance.